RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc

```diff
@@ -219,6 +219,16 @@ int RecHitTools::getSiThickIndex(const DetId& id) const {
   return thickIndex;
 }
 
+std::pair<float, float> RecHitTools::getScintDEtaDPhi(const DetId& id) const {
+  if (!isScintillator(id)) {
+    LogDebug("getScintDEtaDPhi::InvalidSintDetid")
+        << "det id: " << std::hex << id.rawId() << std::dec << ":" << id.det() << " is not HGCal scintillator!";
+    return std::pair<float, float>(0., 0.);
+  }
+  auto geom = static_cast<const HGCalGeometry*>(getSubdetectorGeometry(id));
+  return std::pair<float, float>(geom->getGeometry(id)->etaSpan(), geom->getGeometry(id)->phiSpan());
```

Comment:
why is the cast needed here?
`getGeometry` together with`etaSpan` and `phiSpan`  are all virtual.

```suggestion
  auto cellGeom = getSubdetectorGeometry(id)->getGeometry(id);
  return {cellGeom->etaSpan(), cellGeom->phiSpan()};
```